### PR TITLE
[KOGITO-4422] clean up kogito code after changes to kie API

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

**Review by danielezonca — APPROVED** [review]

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins retest this

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/RuleUnitGenerator.java [hunk 1]
```diff
@@ -63,7 +64,7 @@
     private RuleUnitConfig config;
     private Collection<QueryModel> queries;
 
-    public RuleUnitGenerator( KogitoBuildContext context, KogitoRuleUnitDescription ruleUnit, String generatedSourceFile) {
+    public RuleUnitGenerator( KogitoBuildContext context, RuleUnitDescription ruleUnit, String generatedSourceFile) {
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/RuleUnitGenerator.java:67]: If you keep `KogitoRuleUnitDescription` here you can avoid the downcast to `AbstractRuleUnitDescriptor` (`getConfig` is defined in `KogitoRuleUnitDescription` interface)

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java [hunk 2]
```diff
@@ -387,15 +387,15 @@ private void generateSessionUnits( List<GeneratedFile> generatedFiles ) {
         }
     }
 
-    private void addUnitConfToKieModule(KogitoRuleUnitDescription ruleUnitDescription) {
+    private void addUnitConfToKieModule(RuleUnitDescription ruleUnitDescription) {
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java:390]: If you keep `KogitoRuleUnitDescription` here you can avoid the downcast to `AbstractRuleUnitDescriptor` (`getConfig` is defined in `KogitoRuleUnitDescription` interface)

**Pull request merged**